sonar-plugin-api/src/main/java/org/sonar/api/batch/SonarIndex.java

```diff
@@ -26,18 +26,16 @@
 import org.sonar.api.resources.Project;
 import org.sonar.api.resources.Resource;
 import org.sonar.api.rules.Violation;
-import org.sonar.graph.DirectedGraphAccessor;
 
 import javax.annotation.CheckForNull;
 
 import java.util.Collection;
-import java.util.Set;
 
 /**
  * @deprecated since 4.5.2 should not be used by plugins. Everything should be accessed using {@link SensorContext}.
  */
 @Deprecated
-public abstract class SonarIndex implements DirectedGraphAccessor<Resource, Dependency> {
+public abstract class SonarIndex {
```

Comment:
![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)